docs/script.md

```diff
@@ -320,290 +502,348 @@ In the above snippet the `task.cpus` holds the value for the {ref}`cpus directiv
 
 See {ref}`Process directives <process-directives>` for details.
 
-(script-closure)=
-
-## Closures
+(implicit-functions)=
 
-Briefly, a closure is a block of code that can be passed as an argument to a function. Thus, you can define a chunk of code and then pass it around as if it were a string or an integer.
+## Implicit functions
 
-More formally, you can create functions that are defined as *first-class objects*.
+The following functions are available in Nextflow scripts:
 
-```groovy
-square = { it * it }
-```
+`branchCriteria( closure )`
+: Create a branch criteria to use with the {ref}`operator-branch` operator.
 
-The curly brackets around the expression `it * it` tells the script interpreter to treat this expression as code. The `it` identifier is an implicit variable that represents the value that is passed to the function when it is invoked.
+`cacheableDir( key )`
+: Create a unique folder for the given key. The same folder name is always returned for the same key.
 
-Once compiled the function object is assigned to the variable `square` as any other variable assignments shown previously. Now we can do something like this:
+`cacheableFile( key, name = null )`
+: Create a unique file name for the given key. The same file name is always returned for the same key. If `name` is not specified, it will be generated from the key.
 
-```groovy
-println square(9)
-```
+`error( message = null )`
+: Throw a script runtime error with an optional error message.
 
-and get the value 81.
+`exit( exitCode = 0, message = null )`
+: Stop the pipeline execution and return an exit code and optional error message.
 
-This is not very interesting until we find that we can pass the function `square` as an argument to other functions or methods. Some built-in functions take a function like this as an argument. One example is the `collect` method on lists:
+`fasta( obj )`
```

Comment:
This has always been an experiment but I think it should be removed and only keep at channel level because it can create a blocking point in the script execution